### PR TITLE
Move gameplay geometry

### PR DESCRIPTION
**Pull request by kfu02**
# Description

This PR has two main changes:
1) Moved the geometry-related code in gameplay_tick() of `gameplay_node` to a few more well-named functions. 
2) Renamed the goalZoneObstacles ROS topic to "defAreaObstacles", which stands for defense area, [the official name](https://robocup-ssl.github.io/ssl-rules/sslrules.html#_defense_area). Updated the header files and the Field object in `stp/rc.py` to match. I thought it would be clearer for someone searching through the rules; let me know if this is overly pedantic.

I also added some comments throughout for clarity.

# Steps to test
Launch sim.

Expected result: As before, the obstacles for the defense area and the goal's walls are present. (Shown as translucent red boxes.) On stop situations the ball should also be considered an obstacle. 

**This last part is untested because I have not correctly configured external ref, so I can't issue a stop command.**

### CODE REVIEW

File: rj_gameplay/rj_gameplay/gameplay_node.py [hunk 1]
```diff
@@ -176,92 +176,109 @@ def gameplay_tick(self) -> None:
...
-                ]),
-                geo_msg.Rect(pt=[
-                    geo_msg.Point(x=-self.field.goal_width_m / 2, y=self.field.length_m + self.field.goal_depth_m),
-                    geo_msg.Point(x=-self.field.goal_width_m / 2 - physical_goal_board_width, y=self.field.length_m),
-                ]),
-            ]
+
+            def_area_obstacles = geo_msg.ShapeSet()
+            self.add_def_areas_to_obs(def_area_obstacles, game_info)
+            self.def_area_obstacles_pub.publish(def_area_obstacles)
```
Comment (kfu02) [inline rj_gameplay/rj_gameplay/gameplay_node.py:194]: This doesn't publish the def_area_obstacles (former "goal_zone_obstacles")

File: rj_gameplay/rj_gameplay/gameplay_node.py [hunk 2]
```diff
@@ -176,92 +176,109 @@ def gameplay_tick(self) -> None:
...
-                    global_obstacles.circles.append(
-                        geo_msg.Circle(center=geo_msg.Point(x=ball_point[0], y=ball_point[1]), radius=0.1))
-                if game_info.is_free_placement():
-                    for t in np.linspace(0.0, 1.0, 20):
-                        placement = game_info.ball_placement()
-
-                        pt = ball_point * t + (1 - t) * placement
-                        global_obstacles.circles.append(
-                            geo_msg.Circle(center=geo_msg.Point(x=pt[0], y=pt[1]), radius=0.8))
+            self.add_goals_to_global_obs(global_obstacles, game_info)
```
Comment (kfu02) [inline rj_gameplay/rj_gameplay/gameplay_node.py:197]: These lines **do** publish the global obs

File: rj_gameplay/rj_gameplay/tactic/goalie_tactic.py [hunk 3]
```diff
@@ -106,9 +106,9 @@ def get_requests(self, world_state: rc.WorldState,
         # TODO: this calculation is copy-pasted from wall_tactic
         # put into common param file: https://www.geeksforgeeks.org/global-keyword-in-python/
 
-        # dist is slightly greater than penalty box bounds
-        box_w = world_state.field.penalty_long_dist_m
-        box_h = world_state.field.penalty_short_dist_m
+        # dist is slightly greater than def_area box bounds
+        box_w = world_state.field.def_area_long_dist_m
```
Comment (kfu02) [inline rj_gameplay/rj_gameplay/tactic/goalie_tactic.py:112]: This works so the problem doesn't lie in how I changed the Field class in stp/rc.py

**Review by kfu02 — COMMENTED** [review]
I got too ambitious with my refactoring. Tried to rename `goal_zone_obstacles` to `def_area_obstacles` as that's what the rulebook terms it; broke those pubs.

Also reorganized gameplay_node to remove the geometry calculations from the tick() method, which worked fine for global_obstacles but not the goal_zone_obstacles. Should've tested the separation before renaming but I did not. 

**EDIT: fixed**

**Comment by kfu02** [comment]
Fixed, thanks Kyle.

**Review by kasohrab — APPROVED** [review]
Looks good to me.

**Review by adithya-gv — APPROVED** [review]
Looks good to me

**Pull request merged**